### DISCUSSION TITLE
Search entry points to optimize YOLO on MIC-710AIX

### DISCUSSION THREAD

**Discussion by pedefe** [ultralytics/ultralytics#16519] [General]
Hello,
I started since few times with YOLO8/Python3 on Windows, and I succeed to test "training" and "tracking".
Now, I try to execute the same programs on ADVANTECH MIC-710AIX, and the execution switch from 8 fps on windows to 1.5 fps on MIC.
I think that my programs don't use Nvidia components, and it can be the first reason.
How to use  Nvidia elements ?
Best regards.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @pedefe, thank you for reaching out and using Ultralytics 🚀! It sounds like you're working on optimizing YOLO performance on the ADVANTECH MIC-710AIX.

For getting started, we recommend checking out the [Docs](https://docs.ultralytics.com), where you'll find valuable resources, including [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples. These can be helpful in ensuring you're using NVIDIA components correctly.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us debug it effectively.

For custom optimizations or if you have any training ❓ questions, please provide detailed information, including your setup and configuration. This will help us guide you better.

Feel free to join the Ultralytics community for more support. Whether it's on [Discord](https://ultralytics.com/discord) 🎧 for real-time chat, [Discourse](https://community.ultralytics.com) for detailed discussions, or on our [Subreddit](https://reddit.com/r/ultralytics) for community knowledge sharing.

## Upgrade

Ensure you have the latest version of the `ultralytics` package, including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml), in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLOv8 can be run in the following environments with everything preinstalled:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** and **Amazon** options
- **Docker Image** for seamless integration <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

An Ultralytics engineer will assist you soon to provide more detailed help. If you have any more questions, feel free to ask! 😊

**Comment by glenn-jocher** [comment]
@pedefe to utilize NVIDIA components on the ADVANTECH MIC-710AIX, ensure that CUDA and cuDNN are properly installed and configured. You might also consider exporting your model to TensorRT for improved performance. For more details, visit our NVIDIA Jetson guide.

**Reply by pedefe** [reply]
Thanks.

I have read a lot of tutorial with your suggest.
Now I am trying to install necessary components to execute YOLO applications in Python3 on the MIC-710AIX, but all steps fail.

- "sudo apt install nvidia-cuda-toolkit" : no installable version 
- "sudo ubuntu-drivers autoinstall" : No drivers found for installation
The Ubuntu release is 20.04.6
Any other suggest ?
Best regards.

**Reply by glenn-jocher** [reply]
@pedefe it seems like there might be compatibility issues with your Ubuntu version. Please ensure your system is updated and consider checking NVIDIA's official site for specific driver and CUDA toolkit versions compatible with your hardware.
